### PR TITLE
[Model]PPI dataloader and inductive learning script.

### PR DESCRIPTION
**Pull request by sufeidechabei**
## Changes
1. PPI dataloader.
2. Training script.

### CODE REVIEW

File: examples/pytorch/gat/train_ppi.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+"""
+
+
+import numpy as np
+import torch
+import dgl
+import torch.nn.functional as F
+import argparse
+from sklearn.metrics import f1_score
+from train import GAT
```
Comment (jermainewang) [inline examples/pytorch/gat/train_ppi.py]: Put `GAT` in `gat.py`. Let `train.py` and `train_ppi.py` import from it.

File: python/dgl/data/ppi.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    with open('{}/ppi/ppi-class_map.json'.format(dir)) as jsonfile:
+        class_map = json.load(jsonfile)
+    features = np.load('{}/ppi/ppi-feats.npy'.format(dir))
+    label_list = []
+    for i in range(len(class_map)):
+        label_list.append(np.expand_dims(np.array(class_map[str(i)]), axis=0))
+    labels = np.concatenate(label_list)
+    graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
+    graph_id = np.load('{}/ppi/graph_id.npy'.format(dir))
```
Comment (jermainewang) [inline python/dgl/data/ppi.py]: Put the above codes in `__init__`.

File: python/dgl/data/ppi.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        Paramters
+        ---------
+        mode : str
+            ('train', 'valid', 'test').
+        """
+        self.mode = mode
+        self._preprocess()
+        self._normalize()
+    def _preprocess(self):
```
Comment (jermainewang) [inline python/dgl/data/ppi.py]: Add a newline above.

File: python/dgl/data/ppi.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        if self.mode == 'test':
+            self.test_mask_list = []
+            self.test_graphs = []
+            self.test_labels = []
+            for test_graph_id in range(23, 25):
+                test_graph_mask = np.where(self.graph_id == test_graph_id)[0]
+                self.test_mask_list.append(test_graph_mask)
+                self.test_graphs.append(self.graph.subgraph(test_graph_mask))
+                self.test_labels.append(self.labels[test_graph_mask])
+    def _normalize(self):
```
Comment (jermainewang) [inline python/dgl/data/ppi.py]: Add a newline above.

File: python/dgl/data/ppi.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        """
+        Normalize the features
+        """
+
+        train_feats = self.features[np.concatenate(self.train_mask_list)]
+        scaler = StandardScaler()
+        scaler.fit(train_feats)
+        self.features = scaler.transform(self.features)
+    def __len__(self):
```
Comment (jermainewang) [inline python/dgl/data/ppi.py]: Add a newline above.

File: python/dgl/data/ppi.py (hunk 2 above)
Comment (sufeidechabei) [reply python/dgl/data/ppi.py]: If I put this part into `__init__`, it will load the data three times when create the train, test, validation and spend more time.

File: python/dgl/data/ppi.py (hunk 5 above)
Comment (sufeidechabei) [reply python/dgl/data/ppi.py]: I have done them.

File: python/dgl/data/ppi.py (hunk 2 above)
Comment (jermainewang) [reply python/dgl/data/ppi.py]: My reply is accidentally cancelled. Have you received it?

Comment (jermainewang) [reply python/dgl/data/ppi.py]: You are using class attributes in a wrong way. Class attributes will be evaluated with the class definition.
```python
class C:
  x = 1
  print('xxx')
# 'xxx' will be printed immediately after the above definition.
print(C.x)  # x become an attribute of the class not an instance
```

This means if user import `dgl.data.ppi`, the dataset will be loaded no matter what. User cannot control it. Even worse, if `from ppi import PPIDataset` is in `dgl.data.__init__`, then a user who wants to load other dataset will accidentally load PPIDataset.

There are many ways to fix it:
* Load the whole dataset and split the train/val/test sets outside the dataloader. I believer there should be clean ways to achieve that.
* Split the dataset beforehand.
* Add an `offset` argument in the `init` function to skip graphs before.

Comment (sufeidechabei) [reply python/dgl/data/ppi.py]: I have add the `load_ppi` function to load the data, `PPIDataset` is used for preprocessing the dataset.

Comment (jermainewang) [reply python/dgl/data/ppi.py]: That's not a good practice. Please have a look at [Pytorch's MNIST dataset](https://pytorch.org/docs/0.4.0/_modules/torchvision/datasets/mnist.html). They also have train and test set. See how they use one Dataset API to solve the problem.

**Comment by jermainewang** [comment]
Thanks @sufeidechabei !
